packages/core/bootstrap/src/lib/util.ts

```diff
@@ -483,32 +513,103 @@ export const registerUnhandledRejectionHandler = (): void => {
+    identity?: {
+      sourceIp: string
+    }
+  }
+}
 /**
- * Remove stale request entries from an array.
- * This function assumes that the array is sorted by timestamp,
- * where the oldest entry lives in the 0th index, and the newest entry
- * lives in the arr.length-1th index
- * @param items The items to filter
- * @param filter The windowing function to apply
+ * Get client IP address.
+ *
+ * @param req
+ * @returns {string} ip - The IP address if known, defaulting to 'unknown'.
  */
-export function sortedFilter<T>(items: T[], windowingFunction: (item: T) => boolean): T[] {
-  // if we want a higher performance implementation
-  // we can later resort to a custom array class that is circular
-  // so we can amortize expensive operations like resizing, and make
-  // operations like moving the head index much quicker
-  const firstNonStaleItemIndex = items.findIndex(windowingFunction)
-  if (firstNonStaleItemIndex === -1) {
-    return []
+export function getClientIp(request: unknown): string {
+  const req = request as IncomingHttpRequest
+  if (req.headers) {
+    if (isIp(req.headers['x-client-ip'])) return req.headers['x-client-ip'] // Standard headers used by Amazon EC2, Heroku, and others.
+    const xForwardedFor = getClientIpFromXForwardedFor(req.headers['x-forwarded-for']) // Load-balancers (AWS ELB) or proxies.
+    if (isIp(xForwardedFor)) return String(xForwardedFor)
+    if (isIp(req.headers['cf-connecting-ip'])) return req.headers['cf-connecting-ip'] // Cloudflare.
+    if (isIp(req.headers['fastly-client-ip'])) return req.headers['fastly-client-ip'] // Fastly and Firebase hosting header
+    if (isIp(req.headers['true-client-ip'])) return req.headers['true-client-ip'] // Akamai and Cloudflare: True-Client-IP.
+    if (isIp(req.headers['x-real-ip'])) return req.headers['x-real-ip'] // Default nginx proxy/fcgi; alternative to x-forwarded-for, used by some proxies.
+    if (isIp(req.headers['x-cluster-client-ip'])) return req.headers['x-cluster-client-ip'] // Rackspace LB and Riverbed's Stingray
+    if (isIp(req.headers['x-forwarded'])) return req.headers['x-forwarded']
+    if (isIp(req.headers['forwarded-for'])) return req.headers['forwarded-for']
+    if (isIp(req.headers.forwarded)) return req.headers.forwarded
   }
 
-  return items.slice(firstNonStaleItemIndex)
+  // Remote address checks
+  if (req.connection) {
+    if (isIp(req.connection.remoteAddress)) return req.connection.remoteAddress
+    if (req.connection.socket && isIp(req.connection.socket.remoteAddress))
+      return req.connection.socket.remoteAddress
+  }
+
+  if (req.socket && isIp(req.socket.remoteAddress)) return req.socket.remoteAddress
+  if (req.info && isIp(req.info.remoteAddress)) return req.info.remoteAddress
+
+  // AWS Api Gateway + Lambda
+  if (
+    req.requestContext &&
+    req.requestContext.identity &&
+    isIp(req.requestContext.identity.sourceIp)
+  )
+    return req.requestContext.identity.sourceIp
+
+  return 'unknown'
```

Comment:
I assume you mean we could set 'trust proxy'? I wanted to avoid that in case it had any impact on our infra setup, but if  you think it's ok, I'm happy to do it that way 👍